src/TSTransformer/util/offset.ts

```diff
@@ -16,6 +16,20 @@ export function offset(expression: luau.Expression, value: number) {
 	if (value === 0) {
 		return expression;
 	}
+	// special case to handle adding and removing the offset
+	if (expression.kind === luau.SyntaxKind.BinaryExpression) {
```

Comment:
This triple nested if can and probably should just be a single if with `binary && literal offset && (+ || -)`